strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+  * _gentleUpTiles:_ Indicates how many tiles gently slope upwards (like in Speed Booster Hall).
+  * _gentleDownTiles:_ Indicates how many tiles gently slope downwards (like in Speed Booster Hall).
+  * _steepUpTiles:_ Indicates how many tiles steeply slope upwards (like in Landing Site).
+  * _steepDownTiles:_ Indicates how many tiles steeply slope downwards (like in Landing Site).
+  * _startingDownTiles:_  Indicates how many tiles slope downwards at the expected start of the running space. A stutter can't be executed on those tiles.
+
+In most cases, what matters is the effective runway length (ERL), which takes into account how slopes temporarily slow down Samus' horizontal movement:
+
+`ERL = length - startingDownTiles - 9/16 * (1 - openEnd) + 1/3 * steepUpTiles + 1/7 * steepDownTiles + 5/27 * gentleUpTiles + 5/59 * gentleDownTiles`
+
+There is some variance in how much down slopes slow down Samus' movement, depending on specific alignment of Samus' X position. The amount of variance is small enough to be neglected as long as some small amount of lenience is included in how runways are required to be used.
+
+In a heated room, a `leaveWithRunway` exit condition implicitly includes `heatFrames` needed to use the runway. The amount of `heatFrames` required depends on the entrance condition in the next room, and the details of how these `heatFrames` may be calculated are described under each entrance condition [below](#entrance-conditions). If the `from` and `to` nodes of the strat are different nodes, then it is assumed that the strat property `requires` already includes any heat frames needed to reach the starting point of the runway (the side furthest from the door), in which case the implicit `heatFrames` in `leaveWithRunway` will only account for the heat frames needed to use the runway in one direction, moving towards the door. If the `from` and `to` nodes of the strat are the same node, then the implicit `heatFrames` includes all heat frames needed to position appropriately on the runway (starting from the door), execute the required movement, and exit the room through the door.
+
+When a `leaveWithRunway` condition is used to charge a shinespark, it implicitly includes a `canShinecharge` requirement with `usedTiles` based on the runway length (and also based on a continuation of the runway in the destination room where applicable). Again, this is described in more detail under the applicable entrance conditions below.
+
+When a `leaveWithRunway` conditions occurs on a door in a water environment, it implicitly includes a `Gravity` requirement.
```

Comment:
Yeah East Pants Room could maybe use a rework; I think it would work better if it were just treated as a separate room with its own nodes, instead of how it is now where there's a fake "transition" from Pants Room to East Pants Room when going through Pants Room. So then the door environment on the right door of Pants Room and East Pants Room could be different from each other (one of them water and one of them air).